frontend/src/routes/Video.tsx

```diff
@@ -368,14 +422,33 @@ const VideoDate: React.FC<VideoDateProps> = ({ event }) => {
         ? null
         : new Date(event.syncedData.endTime);
 
+    return {
+        created,
+        updated,
+        startTime,
+        endTime,
+        hasStarted: startTime != null && startTime < new Date(),
+        hasEnded: endTime != null && endTime < new Date(),
+    };
+};
+
+type VideoDateProps = {
+    event: SyncedEvent;
+};
+
+const VideoDate: React.FC<VideoDateProps> = ({ event }) => {
+    const { t, i18n } = useTranslation();
+
+    const { created, updated, startTime, endTime, hasStarted, hasEnded } = getEventTimeInfo(event);
+
     const fullOptions = { dateStyle: "long", timeStyle: "short" } as const;
     let inner;
-    if (event.isLive && endTime && endTime < new Date()) {
+    if (event.isLive && endTime && hasEnded) {
```

Comment:
conditions like this one are rather frequent in this patch, I wonder whether one can simplify them, and/or put them in a function. Maybe just add fields to `getEventTimeInfo`? :thinking: 

Anyway, I feel like (repeatedly) having to check `endTime && hasEnded` is error prone.

Reply:
The problem is that returning booleans from `getEventTimeInfo` does not tell typescript that `endTime` is non-null. And `endTime` is often used in the if bodies, for example. Sure, one could build a complex type there but mh. Dunno. Haven't come up with a super good solution.

Reply:
Mh, fair. I'ma go on record and say "I don't like it!" but ok, I guess no need to burden this review with it if it's more complex. :disappointed: